src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.Factory.cs

```diff
@@ -0,0 +1,216 @@
+        /// <param name="lengths"></param>
+        /// <returns></returns>
+        public static Tensor<T> CreateUninitialized<T>(bool mustPin, ReadOnlySpan<nint> lengths)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanNDHelpers.CalculateTotalLength(lengths);
+            T[] values = GC.AllocateUninitializedArray<T>((int)linearLength, mustPin);
+            return new Tensor<T>(values, lengths.ToArray(), mustPin);
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="mustPin"></param>
+        /// <param name="lengths"></param>
+        /// <param name="strides"></param>
+        /// <returns></returns>
+        public static Tensor<T> CreateUninitialized<T>(bool mustPin, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanNDHelpers.CalculateTotalLength(lengths);
+            T[] values = GC.AllocateUninitializedArray<T>((int)linearLength, mustPin);
+            return new Tensor<T>(values, lengths.ToArray(), strides.ToArray(), mustPin);
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="address"></param>
+        /// <param name="lengths"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static unsafe SpanND<T> CreateSpan<T>(T* address, ReadOnlySpan<nint> lengths) => new SpanND<T>(address, lengths, false);
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="address"></param>
+        /// <param name="lengths"></param>
+        /// <param name="strides"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static unsafe SpanND<T> CreateSpan<T>(T* address, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides) => new SpanND<T>(address, lengths, false, strides);
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="address"></param>
+        /// <param name="lengths"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static unsafe ReadOnlySpanND<T> CreateReadOnlySpan<T>(T* address, ReadOnlySpan<nint> lengths) => new ReadOnlySpanND<T>(address, lengths, false);
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="address"></param>
+        /// <param name="lengths"></param>
+        /// <param name="strides"></param>
+        /// <returns></returns>
+        [CLSCompliant(false)]
+        public static unsafe ReadOnlySpanND<T> CreateReadOnlySpan<T>(T* address, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides) => new ReadOnlySpanND<T>(address, lengths, false, strides);
```

Comment:
nit: Some of these lines are pretty long and the `=> ...` blends in with the rest of the signature. I find it helpful to break it up into two lines like this (or similar, or just do standard braces and `return ...`)
```suggestion
        public static unsafe ReadOnlySpanND<T> CreateReadOnlySpan<T>(T* address, ReadOnlySpan<nint> lengths, ReadOnlySpan<nint> strides)
            => new ReadOnlySpanND<T>(address, lengths, false, strides);
```

Reply:
So at least these 2 have been removed since we didn't need them here.